test/issuance.spec.ts

```diff
@@ -0,0 +1,322 @@
+  const fixtureWithContract = fixtures.withContract;
+  const entropy31bytes = Buffer.from(
+    '2b73af1c9ae64a6903b3055361dd7b75082003a85374049982fc1e8f31b9a8',
+    'hex',
+  );
+  const prevout: issuance.OutPoint = {
+    txHash: Buffer.from(fixture.prevout.txHash, 'hex').reverse(),
+    vout: fixture.prevout.index,
+  };
+
+  describe('Issuance artifacts generation (entropy, asset value and token value)', () => {
+    describe('Entropy generation', () => {
+      it('should properly generate the entropy from a prevout point of the blockchain', () => {
+        const entropy = issuance.generateEntropy(prevout);
+        assert.strictEqual(entropy.toString('hex'), fixture.expectedEntropy);
+      });
+    });
+
+    describe('Asset calculation', () => {
+      it('should compute the asset value from an entropy previously generated', () => {
+        const asset = issuance.calculateAsset(
+          Buffer.from(fixture.expectedEntropy, 'hex'),
+        );
+        assert.strictEqual(
+          asset.reverse().toString('hex'),
+          fixture.expectedAsset,
+        );
+      });
+
+      it('should throw an error if the entropy has not a lenght of 32 bytes', () => {
+        assert.throws(() => issuance.calculateAsset(entropy31bytes));
+      });
+    });
+
+    describe('Token calculation', () => {
+      it('should compute the reissuance token value from an entropy previously generated', () => {
+        const token = issuance.calculateReissuanceToken(
+          Buffer.from(fixture.expectedEntropy, 'hex'),
+        );
+        assert.strictEqual(
+          token.reverse().toString('hex'),
+          fixture.expectedToken,
+        );
+      });
+
+      it('should throw an error if the entropy has not a lenght of 32 bytes', () => {
+        assert.throws(() => issuance.calculateReissuanceToken(entropy31bytes));
+      });
+    });
+  });
+
+  describe('Issuance object generation', () => {
+    function validate(i: Issuance): boolean {
+      try {
+        typeforce(types.Hash256bit, i.assetBlindingNonce);
+        typeforce(types.Hash256bit, i.assetEntropy);
+        typeforce(
+          types.oneOf(
+            types.ConfidentialValue,
+            types.ConfidentialCommitment,
+            types.BufferOne,
+          ),
+          i.assetAmount,
+        );
+        typeforce(
+          types.oneOf(
+            types.ConfidentialValue,
+            types.ConfidentialCommitment,
+            types.BufferOne,
+          ),
+          i.tokenAmount,
+        );
+        return true;
+      } catch (err) {
+        return false;
+      }
+    }
+
+    it('should create a correct Issuance object without an issuance contract', () => {
+      if (!fixture.prevout) throw new Error('no prevout in issuance.json');
+      const iss: Issuance = issuance.newIssuance(10, 22);
+      assert.strictEqual(validate(iss), true);
+    });
+
+    it('should create a valid Issuance object with an issuance contract', () => {
```

Comment:
same here, but we have to check that iss.contract is not undefined

Reply:
The issuance interface does not contain a contract member. Nevertheless, I can check if the asset entropy is empty (i.e 0 x 32 bits) when the asset contract is undefined.